imagetest/utils/test_utils.go

```diff
@@ -199,6 +199,33 @@ func GetProjectZone(ctx context.Context) (string, string, error) {
 	return project, zone, nil
 }
 
+// GetInstanceZoneProject gets the instance name, zone, and project id strings
+func GetInstanceZoneProject() (string, string, string, error) {
+	var fqdnString string
+	if runtime.GOOS == "windows" {
+		procStatus, err := RunPowershellCmd("Invoke-RestMethod -Headers @{'Metadata-Flavor' = 'Google'} -Uri \"http://metadata.google.internal/computeMetadata/v1/instance/hostname\"")
+		if err != nil {
+			return "", "", "", fmt.Errorf("failed to get project, zone, or instance on windows: %v", err)
+		}
+		fqdnString = strings.TrimSpace(procStatus.Stdout)
+	} else {
+		fqdnBytes, err := exec.Command("curl", "http://metadata.google.internal/computeMetadata/v1/instance/hostname", "-H", "Metadata-Flavor: Google").Output()
```

Comment:
This mechanism seems kind of fragile, since it relies on DNS naming conventions. It will break if project level DNS is not set to zonal or if the instance edits DNS settings or uses a custom hostname, which is something we do in some tests. We also have `instance/zone`, `project/project-id`, and `instance/name` keys to tell use these things authoritatively.